test/e2e/scheduling/hard_constraints.go

```diff
@@ -915,6 +919,285 @@ var _ = SIGDescribe("SchedulingHardConstraints [Serial]", func() {
+	// 7. Make sure the antiAffinityPod is scheduled to the second nmnode.
+	// 8. Create one failedPod with label[1] that needs 40% of the extra resource.
+	// 9. Make sure the failedPod is not scheduled becaues it only can be scheduled to the first nmnode due to antiAffinityPod's anti affinity.
+	//    But the first nmnode haven't enough resource.
+	// 10. Create one successPod with label[0] that needs 40% of the extra resource.
+	// 11. Make sure the successPod is scheduled to the second nmnode, because it isn't affected by any inter-pod affinity.
+	/*
+		Testname: Scheduling pods with inter-pod affinity matching for nmnodes
+		Description: Scheduling MUST meet inter-pod affinity requirements and scheduling pods MUST fail if no resource meets the specified pod
+	*/
+	ginkgo.Context("validates pod scheduling fits inter-pod affinity[requiredDuringSchedulingIgnoredDuringExecution] for nmnodes", func() {
+		testLabelKeys := []string{"godel.bytedance.com/test-label1", "godel.bytedance.com/test-label2"}
+		testLabelValues := []string{"test", "foo"}
+		var beardsecond v1.ResourceName = "example.com/beardsecond"
+		var nodeNames []string
+
+		ginkgo.BeforeEach(func() {
+			WaitForStableCluster(cs, workerNodes)
+			ginkgo.By("cluster is stable")
+			nodeNames = Get2NodesThatCanRunPod(f)
+
+			ginkgo.By("Create nmnodes and set fake resource for " + nodeNames[0] + " and " + nodeNames[1])
+			// Get node object:
+			for _, testNodeName := range nodeNames {
+				node, err := cs.CoreV1().Nodes().Get(context.TODO(), testNodeName, metav1.GetOptions{})
+				framework.ExpectNoError(err, "unable to get node object for node %v", testNodeName)
+
+				nmnodeTemplate := GetNMNodeTemplateByNode(node)
+				(*nmnodeTemplate.Status.ResourceCapacity)[beardsecond] = resource.MustParse("1000")
+				(*nmnodeTemplate.Status.ResourceAllocatable)[beardsecond] = resource.MustParse("1000")
+				_, err = fs.NodeV1alpha1().NMNodes().Create(context.TODO(), nmnodeTemplate, metav1.CreateOptions{})
+				framework.ExpectNoError(err, "unable to create NM node for node %v", testNodeName)
+			}
+		})
+
+		ginkgo.AfterEach(func() {
+			ginkgo.By("Remove nmnodes")
+			for _, testNodeName := range nodeNames {
+				// remove nmnodes:
+				if testNodeName != "" {
+					err := fs.NodeV1alpha1().NMNodes().Delete(context.TODO(), testNodeName, metav1.DeleteOptions{})
+					framework.ExpectNoError(err, "unable to remove nmnode %v", testNodeName)
+				}
+			}
+		})
+
+		ginkgo.It("verify inter-pod affinity matches pod labels for nmnodes", func() {
+			ginkgo.By("Trying to create basicPod, ns: " + ns)
+			basicPodConf := pausePodConfig{
+				Name:        "basic-pod-" + string(uuid.NewUUID()),
+				Annotations: WithNMNodeLauncher(WithHardConstraints(GetPodAnnotations(podutil.GuaranteedPod, podutil.Kubelet), []string{interpodaffinity.Name})),
+				Labels:      map[string]string{testLabelKeys[0]: testLabelValues[0], testLabelKeys[1]: testLabelValues[1]},
```

Comment:
These nodes here all have the node manager launcher right? So it seems that directly replacing`podutil.Kubelet` with `podutil.NodeManager` will be fine?